### PR TITLE
Feature/enhance procedure

### PR DESCRIPTION
**Pull request by AmrSaleh2234**
## 🎯 Summary

-
## 🧾 Related Ticket

- Jira: [ABC-123](https://jira.your-company.com/browse/ABC-123)

## 🔄 Type of Change

- [ ] Feature
- [ ] Bugfix
- [ ] Refactor
- [ ] Hotfix
- [ ] Chore / Maintenance

## 🧪 How to Test

1. Call endpoint: `GET /api/...`
2. Use test data: `...`
3. Expected result:
   - 
    -

## ✅ Checklist

- [ ] Performed **self-review** of the code
- [ ] Updated **validations** if business logic changed
- [ ] Updated **API docs / Postman collection** if the endpoint changed
- [ ] All **tests are passing** (if applicable)
- [ ] No **breaking changes** for existing clients

## 🧱 Migration / Database Changes

- [ ] No database changes
- [ ] There is a new migration (describe below):

### CODE REVIEW

File: modules/EmployeeTask/Services/EmployeeTaskApprovalService.php [hunk 1]
```diff
@@ -219,13 +223,37 @@ private function resolveApprovalProcedureSetting(EmployeeTaskRequest $task): ?Pr
...
         );
     }
 
+    /**
+     * Load a specific internal procedure setting by ID, verifying it belongs
+     * to the task's company/category parent and has a form set.
+     */
+    private function loadInternalProcedureSetting(string $id, EmployeeTaskRequest $task): ?ProcedureSetting
+    {
+        $setting = ProcedureSetting::query()
```
Comment (AmrSaleh2234) [inline modules/EmployeeTask/Services/EmployeeTaskApprovalService.php:240]: The query could be optimized by adding `select('*')` to explicitly state which columns are needed rather than relying on the default select all behavior.

File: modules/EmployeeTask/Services/EmployeeTaskApprovalService.php [hunk 2]
```diff
@@ -219,13 +223,37 @@ private function resolveApprovalProcedureSetting(EmployeeTaskRequest $task): ?Pr
...
+            ->where('id', $id)
+            ->whereNotNull('form')
+            ->whereHas('parent', function ($q) use ($task) {
```
Comment (AmrSaleh2234) [inline modules/EmployeeTask/Services/EmployeeTaskApprovalService.php:243]: The parent relationship query could be simplified by using `whereBelongsTo()` if the relationship is properly defined.

File: modules/EmployeeTask/Services/EmployeeTaskExtensionService.php [hunk 3]
```diff
@@ -190,13 +193,37 @@ private function resolveExtensionProcedureSetting(\Modules\EmployeeTask\Models\E
         $task->loadMissing('user.userProfessionalData');
```
Comment (AmrSaleh2234) [inline modules/EmployeeTask/Services/EmployeeTaskExtensionService.php:193]: The method name `resolveExtensionProcedureSetting` doesn't match the new implementation which resolves procedure settings by form rather than by branch. Consider renaming it to better reflect its new purpose.

File: modules/EmployeeTask/Services/EmployeeTaskExtensionService.php [hunk 4]
```diff
@@ -190,13 +193,37 @@ private function resolveExtensionProcedureSetting(\Modules\EmployeeTask\Models\E
...
         $branchId = $task->user?->userProfessionalData?->branch_id;
 
-        return $this->workflow->resolveProcedureSettingForBranch(
-            ProcedureSettingType::EmployeeTaskExtension->value,
+        return $this->workflow->resolveInternalProcedureSettingByForm(
```
Comment (AmrSaleh2234) [inline modules/EmployeeTask/Services/EmployeeTaskExtensionService.php:196]: The parameter `ProcedureSettingType::EmployeeTask->value` should be validated against `ProcedureSettingType` enum values to ensure type safety.

File: modules/EmployeeTask/Services/EmployeeTaskExtensionService.php [hunk 5]
```diff
@@ -190,13 +193,37 @@ private function resolveExtensionProcedureSetting(\Modules\EmployeeTask\Models\E
...
             $task->company_id,
             $branchId,
         );
     }
 
+    /**
+     * Load a specific internal procedure setting by ID, verifying it belongs
+     * to the task's company/category parent and has a form set.
+     */
+    private function loadInternalProcedureSetting(string $id, \Modules\EmployeeTask\Models\EmployeeTaskRequest $task): ?ProcedureSetting
```
Comment (AmrSaleh2234) [inline modules/EmployeeTask/Services/EmployeeTaskExtensionService.php:208]: The method `loadInternalProcedureSetting` should include validation for the `$id` parameter to ensure it's a valid UUID or ID format before querying.

File: modules/EmployeeTask/Services/EmployeeTaskExtensionService.php [hunk 6]
```diff
@@ -190,13 +193,37 @@ private function resolveExtensionProcedureSetting(\Modules\EmployeeTask\Models\E
...
+            ->where('id', $id)
+            ->whereNotNull('form')
+            ->whereHas('parent', function ($q) use ($task) {
+                $q->where('type', ProcedureSettingType::EmployeeTask->value)
+                  ->where('company_id', $task->company_id);
+            })
+            ->with(['steps' => fn ($q) => $q->orderBy('step_order')])
+            ->first();
+
+        if (! $setting) {
```
Comment (AmrSaleh2234) [inline modules/EmployeeTask/Services/EmployeeTaskExtensionService.php:220]: Consider adding more context to the exception message in `EmployeeTaskException::invalidProcedureSetting()` to help with debugging (e.g., include the setting ID that was not found).

File: modules/EmployeeTask/Services/EmployeeTaskRequestService.php [hunk 7]
```diff
@@ -35,9 +35,9 @@ public function __construct(
 
     public function create(CreateEmployeeTaskRequestDTO $dto): EmployeeTaskRequest
     {
-        $procedureType = ProcedureSettingType::EmployeeTaskRequest->value;
+        $procedureType = ProcedureSettingType::EmployeeTask->value;
         $context       = $dto->projectId ? ['project_id' => $dto->projectId] : [];
-        $preview       = $this->workflow->getApprovalResponsibles($procedureType, $dto->userId, $context);
+        $preview       = $this->workflow->getApprovalResponsibles($procedureType, $dto->userId, $context, InternalProcessForm::StartTask->value);
```
Comment (AmrSaleh2234) [inline modules/EmployeeTask/Services/EmployeeTaskRequestService.php:40]: The added parameter `InternalProcessForm::StartTask->value` should be documented in the method signature of `getApprovalResponsibles` to make it clear what values are expected here.

File: modules/EmployeeTask/Services/EmployeeTaskRequestService.php [hunk 8]
```diff
@@ -117,11 +124,10 @@ public function approve(string $id, string $adminId): EmployeeTaskRequest
 // dd($id, $adminId);
         return DB::transaction(function () use ($id, $adminId, $task): EmployeeTaskRequest {
             $process = Process::query()
-                ->where('processable_type', ProcedureSettingType::EmployeeTaskRequest->value)
+                ->where('processable_type', ProcedureSettingType::EmployeeTask->value)
```
Comment (AmrSaleh2234) [inline modules/EmployeeTask/Services/EmployeeTaskRequestService.php:127]: The change from `ProcedureSettingType::EmployeeTaskRequest` to `ProcedureSettingType::EmployeeTask` seems significant. Ensure this change is intentional and properly tested as it affects the processable type being queried.

File: modules/EmployeeTask/Services/EmployeeTaskRequestService.php [hunk 9]
```diff
@@ -117,11 +124,10 @@ public function approve(string $id, string $adminId): EmployeeTaskRequest
 // dd($id, $adminId);
```
Comment (AmrSaleh2234) [inline modules/EmployeeTask/Services/EmployeeTaskRequestService.php:124]: Remove commented out debug code (`// dd($id, $adminId);`) before merging.

File: modules/EmployeeTask/Services/EmployeeTaskRequestService.php [hunk 10]
```diff
@@ -117,11 +124,10 @@ public function approve(string $id, string $adminId): EmployeeTaskRequest
...
         return DB::transaction(function () use ($id, $adminId, $task): EmployeeTaskRequest {
             $process = Process::query()
-                ->where('processable_type', ProcedureSettingType::EmployeeTaskRequest->value)
+                ->where('processable_type', ProcedureSettingType::EmployeeTask->value)
                 ->where('processable_id', $task->id)
                 ->where('status', ProcessStatus::InProgress)
                 ->firstOrFail();
-// dd($process);
             $step = $this->findPendingStepForActor($process, $adminId);
```
Comment (AmrSaleh2234) [inline modules/EmployeeTask/Services/EmployeeTaskRequestService.php:131]: Remove commented out debug code (`// dd($process);`) before merging.

File: modules/ProcedureSetting/Controllers/ProcedureSettingController.php [hunk 11]
```diff
@@ -182,7 +182,9 @@ private function presentWorkFlow(WorkFlow $workFlow): array
                 ])
                 ->values()
                 ->all(),
-            'procedure-settings' => ProcedureSettingPresenter::collection($workFlow->procedureSettings),
+            'procedure-settings' => ProcedureSettingPresenter::collection(
```
Comment (AmrSaleh2234) [inline modules/ProcedureSetting/Controllers/ProcedureSettingController.php:185]: The change filters procedureSettings by `whereNull('parent_id')` but doesn't explain why this filtering is needed. Consider adding a comment explaining the business logic behind this filtering if it's not obvious from the context.

File: modules/ProcedureSetting/DTO/CreateProcedureSettingDTO.php [hunk 12]
```diff
@@ -16,12 +16,13 @@ public function __construct(
...
         public readonly ?int $escalation_management_hierarchy_id = null,
         public readonly ?string $work_flow_id = null,
+        public readonly ?string $parent_id = null,
     ) {
     }
 
     public function toArray(): array
     {
-        return [
+        $data = [
```
Comment (AmrSaleh2234) [inline modules/ProcedureSetting/DTO/CreateProcedureSettingDTO.php:25]: Unnecessary variable assignment `$data = ` since it's immediately returned. Just keep the direct return statement as it was before.

File: modules/ProcedureSetting/DTO/CreateProcedureSettingDTO.php [hunk 13]
```diff
@@ -31,6 +32,9 @@ public function toArray(): array
             'deadline_hours' => $this->deadline_hours,
             'escalation_management_hierarchy_id' => $this->escalation_management_hierarchy_id,
             'work_flow_id'       => $this->work_flow_id,
+            'parent_id'          => $this->parent_id,
```
Comment (AmrSaleh2234) [inline modules/ProcedureSetting/DTO/CreateProcedureSettingDTO.php:35]: - Missing validation for `parent_id` in the DTO. Should ensure it's either nullable or has proper validation rules.

File: modules/ProcedureSetting/DTO/CreateProcedureSettingDTO.php [hunk 14]
```diff
@@ -31,6 +32,9 @@ public function toArray(): array
...
         ];
+
+        return $data;
```
Comment (AmrSaleh2234) [inline modules/ProcedureSetting/DTO/CreateProcedureSettingDTO.php:38]: - The `return $data;` statement should be aligned with the opening bracket of the array declaration for consistent code style.

File: modules/ProcedureSetting/Database/Seeders/InternalProcedureSettingsSeeder.php [hunk 15]
```diff
@@ -0,0 +1,124 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Modules\ProcedureSetting\Database\Seeders;
+
+use Illuminate\Database\Seeder;
+use Illuminate\Support\Facades\Schema;
```
Comment (AmrSaleh2234) [inline modules/ProcedureSetting/Database/Seeders/InternalProcedureSettingsSeeder.php:8]: The comment about sub-actions being handled via `InternalProcedureSettings` is useful but should be moved to a class-level docblock rather than being inline with the enum values documentation.

File: modules/ProcedureSetting/Database/Seeders/InternalProcedureSettingsSeeder.php [hunk 16]
```diff
@@ -0,0 +1,124 @@
...
+use Illuminate\Support\Str;
+use Modules\Company\CompanyCore\Models\Company;
+use Modules\ProcedureSetting\Enums\ProcedureSettingType;
+use Modules\ProcedureSetting\Models\ProcedureSetting;
```
Comment (AmrSaleh2234) [inline modules/ProcedureSetting/Database/Seeders/InternalProcedureSettingsSeeder.php:12]: The allowed values documentation should be updated to reflect the new enum cases (`employee_task` and `meeting`).

File: modules/ProcedureSetting/Database/Seeders/InternalProcedureSettingsSeeder.php [hunk 17]
```diff
@@ -0,0 +1,124 @@
...
+use Modules\ProcedureSetting\Models\WorkFlow;
+use Modules\Shared\InternalProcessType\Enums\InternalProcessCondition;
+use Modules\Shared\InternalProcessType\Enums\InternalProcessForm;
```
Comment (AmrSaleh2234) [inline modules/ProcedureSetting/Database/Seeders/InternalProcedureSettingsSeeder.php:15]: The enum case names should follow consistent naming conventions - either all PascalCase or all lowercase_with_underscores. Currently there's a mix (`EmployeeTask` vs `ClientRequest`).

File: modules/ProcedureSetting/Repositories/ProcedureSettingRepository.php [hunk 18]
```diff
@@ -96,7 +96,8 @@ public function listByWorkFlow(array $filters = []): \Illuminate\Support\Collect
             ->with([
                 'managementHierarchies:id,name,type,company_id',
                 'procedureSettings' => function ($q) {
-                    $q->orderBy('sort_order')
+                    $q->whereNull('parent_id')
```
Comment (AmrSaleh2234) [inline modules/ProcedureSetting/Repositories/ProcedureSettingRepository.php:99]: The change from `orderBy('sort_order')` to `whereNull('parent_id')` followed by `orderBy('sort_order')` significantly alters the query logic. Please ensure this change is intentional and properly documented in the PR description as it affects which records will be returned.

File: modules/ProcedureSetting/Repositories/ProcedureSettingRepository.php [hunk 19]
```diff
@@ -169,7 +171,8 @@ public function getDefaultWorkFlowByType(string $type): ?WorkFlow
...
```
Comment (AmrSaleh2234) [inline modules/ProcedureSetting/Repositories/ProcedureSettingRepository.php:174]: The `whereNull('parent_id')` condition might unintentionally filter out valid records that should be included. Consider if this aligns with the business requirements or if you need to handle parent/child relationships differently.

File: modules/ProcedureSetting/Repositories/ProcedureSettingRepository.php [hunk 20]
```diff
@@ -169,7 +171,8 @@ public function getDefaultWorkFlowByType(string $type): ?WorkFlow
...
+                      ->orderBy('sort_order')
```
Comment (AmrSaleh2234) [inline modules/ProcedureSetting/Repositories/ProcedureSettingRepository.php:175]: The chaining of `whereNull` and `orderBy` could be made more readable by putting each method call on a new line with proper indentation for better code organization.

File: modules/ProcedureSetting/Repositories/ProcedureSettingRepository.php [hunk 21]
```diff
@@ -169,7 +171,8 @@ public function getDefaultWorkFlowByType(string $type): ?WorkFlow
...
                       ->with(['escalationManagementHierarchy:id,name,type,company_id', 'workFlow:id,name,company_id']);
```
Comment (AmrSaleh2234) [inline modules/ProcedureSetting/Repositories/ProcedureSettingRepository.php:176]: The eager loading relationships (`with`) should be consistent with the rest of the codebase's style - either always on same line or always on new lines.

File: modules/ProcedureSetting/Repositories/ProcedureSettingRepository.php [hunk 22]
```diff
@@ -169,7 +171,8 @@ public function getDefaultWorkFlowByType(string $type): ?WorkFlow
...
                 },
```
Comment (AmrSaleh2234) [inline modules/ProcedureSetting/Repositories/ProcedureSettingRepository.php:177]: The closing parenthesis and comma placement doesn't match the codebase's typical style (usually on same line in Laravel).

File: modules/ProcedureSetting/Repositories/ProcedureSettingRepository.php [hunk 23]
```diff
@@ -169,7 +171,8 @@ public function getDefaultWorkFlowByType(string $type): ?WorkFlow
...
             ])
```
Comment (AmrSaleh2234) [inline modules/ProcedureSetting/Repositories/ProcedureSettingRepository.php:178]: The closing `])` should be properly aligned with the opening `->with([` for better readability.

File: modules/ProcedureSetting/Repositories/ProcedureSettingRepository.php [hunk 24]
```diff
@@ -169,7 +171,8 @@ public function getDefaultWorkFlowByType(string $type): ?WorkFlow
             ->with([
```
Comment (AmrSaleh2234) [inline modules/ProcedureSetting/Repositories/ProcedureSettingRepository.php:171]: The `with` method's opening bracket should be on the same line as the method call to maintain consistency with Laravel's code style.

File: modules/ProcedureSetting/Repositories/ProcedureSettingRepository.php [hunk 25]
```diff
@@ -169,7 +171,8 @@ public function getDefaultWorkFlowByType(string $type): ?WorkFlow
...
                 'managementHierarchies:id,name,type,company_id',
```
Comment (AmrSaleh2234) [inline modules/ProcedureSetting/Repositories/ProcedureSettingRepository.php:172]: The relationship name `managementHierarchies` should be consistent with Laravel's naming conventions (typically singular for belongsTo, plural for hasMany). Verify if this matches your model relationships.

File: modules/ProcedureSetting/Repositories/ProcedureSettingRepository.php [hunk 26]
```diff
@@ -169,7 +171,8 @@ public function getDefaultWorkFlowByType(string $type): ?WorkFlow
...
                 'procedureSettings' => function ($q) {
```
Comment (AmrSaleh2234) [inline modules/ProcedureSetting/Repositories/ProcedureSettingRepository.php:173]: The closure function parameter `$q` should ideally be more descriptive like `$query` to follow Laravel's common practices.

File: modules/ProcedureSetting/Repositories/ProcedureSettingRepository.php (hunk 21 above)
Comment (AmrSaleh2234) [inline modules/ProcedureSetting/Repositories/ProcedureSettingRepository.php:176]: The eager loaded relationships (`escalationManagementHierarchy`, `workFlow`) should be verified if they're using the correct relationship method names as defined in the models.

File: modules/ProcedureSetting/Repositories/ProcedureSettingRepository.php (hunk 23 above)
Comment (AmrSaleh2234) [inline modules/ProcedureSetting/Repositories/ProcedureSettingRepository.php:178]: The final `])` should have proper spacing before it to match PSR-12 coding standards.

File: modules/ProcedureSetting/Repositories/ProcedureSettingRepository.php (hunk 24 above)
Comment (AmrSaleh2234) [inline modules/ProcedureSetting/Repositories/ProcedureSettingRepository.php:171]: Consider using `with` only for relationships that will actually be used in the subsequent code to avoid unnecessary database queries.

File: modules/ProcedureSetting/Repositories/ProcedureSettingRepository.php (hunk 25 above)
Comment (AmrSaleh2234) [inline modules/ProcedureSetting/Repositories/ProcedureSettingRepository.php:172]: The selected fields (`id,name,type,company_id`) should be consistent with what's actually needed - verify if all these fields are used.

File: modules/ProcedureSetting/Repositories/ProcedureSettingRepository.php (hunk 21 above)
Comment (AmrSaleh2234) [inline modules/ProcedureSetting/Repositories/ProcedureSettingRepository.php:176]: The selected fields for `escalationManagementHierarchy` and `workFlow` should be reviewed to ensure they match what's needed in the application to avoid selecting unnecessary columns.

File: modules/ProcedureSetting/Repositories/ProcedureSettingRepository.php (hunk 26 above)
Comment (AmrSaleh2234) [inline modules/ProcedureSetting/Repositories/ProcedureSettingRepository.php:173]: The closure function could potentially be extracted to a model scope if this query pattern is used frequently in the codebase.

File: modules/ProcedureSetting/Repositories/ProcedureSettingRepository.php (hunk 19 above)
Comment (AmrSaleh2234) [inline modules/ProcedureSetting/Repositories/ProcedureSettingRepository.php:174]: The `whereNull('parent_id')` condition should have a comment explaining why parent records are being filtered, unless it's obvious from business context.

File: modules/ProcedureSetting/Repositories/ProcedureSettingRepository.php (hunk 20 above)
Comment (AmrSaleh2234) [inline modules/ProcedureSetting/Repositories/ProcedureSettingRepository.php:175]: The `orderBy('sort_order')` should specify direction (`asc`/`desc`) explicitly to avoid confusion about the default sorting direction.

File: modules/ProcedureSetting/Repositories/ProcedureSettingRepository.php (hunk 21 above)
Comment (AmrSaleh2234) [inline modules/ProcedureSetting/Repositories/ProcedureSettingRepository.php:176]: The eager loading could be optimized by checking if `escalationManagementHierarchy` and `work

File: modules/ProcedureSetting/Requests/CreateInternalProcedureSettingRequest.php [hunk 27]
```diff
@@ -0,0 +1,36 @@
...
+use Illuminate\Validation\Rule;
+use Modules\ProcedureSetting\Enums\ProcedureSettingType;
+use Modules\Shared\InternalProcessType\Enums\InternalProcessCondition;
+use Modules\Shared\InternalProcessType\Enums\InternalProcessForm;
+
+class CreateInternalProcedureSettingRequest extends FormRequest
+{
+    public function rules(): array
+    {
+        return array_merge([
```
Comment (AmrSaleh2234) [inline modules/ProcedureSetting/Requests/CreateInternalProcedureSettingRequest.php:17]: The `array_merge` should be on multiple lines for better readability since it's merging multiple arrays.

File: modules/ProcedureSetting/Requests/CreateInternalProcedureSettingRequest.php [hunk 28]
```diff
@@ -0,0 +1,36 @@
...
+            'name'              => ['sometimes', 'string', 'max:255'],
+            'type'              => ['required', 'string', Rule::in(ProcedureSettingType::values())],
```
Comment (AmrSaleh2234) [inline modules/ProcedureSetting/Requests/CreateInternalProcedureSettingRequest.php:19]: Consider adding validation for the `type` field to ensure it matches one of the `ProcedureSettingType` enum values.

File: modules/ProcedureSetting/Requests/CreateInternalProcedureSettingRequest.php [hunk 29]
```diff
@@ -0,0 +1,36 @@
...
+            'form'              => ['required', 'string', Rule::in(InternalProcessForm::values())],
```
Comment (AmrSaleh2234) [inline modules/ProcedureSetting/Requests/CreateInternalProcedureSettingRequest.php:20]: Consider adding validation for the `form` field to ensure it matches one of the `InternalProcessForm` enum values.

File: modules/ProcedureSetting/Requests/CreateInternalProcedureSettingRequest.php [hunk 30]
```diff
@@ -0,0 +1,36 @@
...
+            'execute_type'      => ['sometimes', 'string', 'in:parallel,sequence'],
```
Comment (AmrSaleh2234) [inline modules/ProcedureSetting/Requests/CreateInternalProcedureSettingRequest.php:21]: The `execute_type` validation should include a check for valid values ('parallel' or 'sequence').

File: modules/ProcedureSetting/Requests/CreateInternalProcedureSettingRequest.php [hunk 31]
```diff
@@ -0,0 +1,36 @@
...
+            'conditions'        => ['sometimes', 'array'],
```
Comment (AmrSaleh2234) [inline modules/ProcedureSetting/Requests/CreateInternalProcedureSettingRequest.php:22]: The `conditions` field should have validation for its elements if they are expected to follow a specific structure.

File: modules/ProcedureSetting/Requests/CreateInternalProcedureSettingRequest.php [hunk 32]
```diff
@@ -0,0 +1,36 @@
...
+            'appears_before_id' => ['nullable', 'uuid', 'exists:procedure_settings,id'],
```
Comment (AmrSaleh2234) [inline modules/ProcedureSetting/Requests/CreateInternalProcedureSettingRequest.php:23]: The `appears_before_id` field should have a validation to ensure it's a valid UUID.

File: modules/ProcedureSetting/Requests/CreateInternalProcedureSettingRequest.php [hunk 33]
```diff
@@ -0,0 +1,36 @@
...
+            'appears_after_id'  => ['nullable', 'uuid', 'exists:procedure_settings,id'],
```
Comment (AmrSaleh2234) [inline modules/ProcedureSetting/Requests/CreateInternalProcedureSettingRequest.php:24]: The `appears_after_id` field should have a validation to ensure it's a valid UUID.

File: modules/ProcedureSetting/Requests/CreateInternalProcedureSettingRequest.php [hunk 34]
```diff
@@ -0,0 +1,36 @@
...
+            'sort_order'        => ['sometimes', 'integer', 'min:0'],
```
Comment (AmrSaleh2234) [inline modules/ProcedureSetting/Requests/CreateInternalProcedureSettingRequest.php:25]: The `sort_order` field should have a validation to ensure it's a positive integer.

File: modules/ProcedureSetting/Requests/CreateInternalProcedureSettingRequest.php [hunk 35]
```diff
@@ -0,0 +1,36 @@
...
+            'percentage'        => ['nullable', 'numeric', 'min:0', 'max:100'],
```
Comment (AmrSaleh2234) [inline modules/ProcedureSetting/Requests/CreateInternalProcedureSettingRequest.php:26]: The `percentage` field should have a validation to ensure it's between 0 and 100.

File: modules/ProcedureSetting/Requests/CreateInternalProcedureSettingRequest.php [hunk 36]
```diff
@@ -0,0 +1,36 @@
...
+            'deadline_days'     => ['nullable', 'integer', 'min:0'],
```
Comment (AmrSaleh2234) [inline modules/ProcedureSetting/Requests/CreateInternalProcedureSettingRequest.php:27]: The `deadline_days` field should have a validation to ensure it's a positive integer.

File: modules/ProcedureSetting/Requests/CreateInternalProcedureSettingRequest.php [hunk 37]
```diff
@@ -0,0 +1,36 @@
...
+            'deadline_hours'    => ['nullable', 'integer', 'min:0'],
```
Comment (AmrSaleh2234) [inline modules/ProcedureSetting/Requests/CreateInternalProcedureSettingRequest.php:28]: The `deadline_hours` field should have a validation to ensure it's a positive integer.

File: modules/ProcedureSetting/Requests/CreateInternalProcedureSettingRequest.php [hunk 38]
```diff
@@ -0,0 +1,36 @@
...
+        ], InternalProcessCondition::validationRulesForForm($this->input('form'), 'conditions'));
```
Comment (AmrSaleh2234) [inline modules/ProcedureSetting/Requests/CreateInternalProcedureSettingRequest.php:29]: The `InternalProcessCondition::validationRulesForForm` method should be documented to explain its purpose and expected return value.

File: modules/ProcedureSetting/Requests/CreateProcedureSettingRequest.php [hunk 39]
```diff
@@ -28,6 +28,7 @@ public function rules(): array
             'time_limit_hours' => 'nullable|integer|min:0',
             'escalation_management_hierarchy_id' => 'nullable|integer|exists:management_hierarchies,id',
             'work_flow_id'       => 'required|uuid|exists:work_flows,id',
+            'parent_id'          => 'required|uuid|exists:procedure_settings,id',
```
Comment (AmrSaleh2234) [inline modules/ProcedureSetting/Requests/CreateProcedureSettingRequest.php:31]: The `parent_id` validation rule is marked as `required` which might break existing functionality if parent_id is not always required. Consider making it `nullable` if there are cases where a procedure setting can exist without a parent.

Comment (AmrSaleh2234) [inline modules/ProcedureSetting/Requests/CreateProcedureSettingRequest.php:31]: The `exists:procedure_settings,id` validation assumes the parent procedure setting exists, but there might be race conditions where the parent is deleted after validation. Consider adding error handling for this case in the business logic.

Comment (AmrSaleh2234) [inline modules/ProcedureSetting/Requests/CreateProcedureSettingRequest.php:31]: Adding `parent_id` as required could potentially create circular references if not properly handled. Ensure there's validation in the business logic to prevent procedure settings from referencing themselves or creating reference loops.

Comment (AmrSaleh2234) [inline modules/ProcedureSetting/Requests/CreateProcedureSettingRequest.php:31]: The `uuid` validation for `parent_id` should be consistent with the `work_flow_id` validation above it - both are referencing IDs but have different validation rules (`uuid` vs `integer`). Consider standardizing these validations.

Comment (AmrSaleh2234) [inline modules/ProcedureSetting/Requests/CreateProcedureSettingRequest.php:31]: The `parent_id` field addition should be reflected in the API documentation since it's a required field that will affect all create requests. Please confirm this has been updated in the API docs.

Comment (AmrSaleh2234) [inline modules/ProcedureSetting/Requests/CreateProcedureSettingRequest.php:31]: Consider adding database-level constraints (foreign key) to match this validation rule to ensure data integrity at the database level, not just application level.

Comment (AmrSaleh2234) [inline modules/ProcedureSetting/Requests/CreateProcedureSettingRequest.php:31]: The change makes `parent_id` required but doesn't show any corresponding changes to handle cases where this might fail (e.g., error messages, API response updates). Please confirm these aspects have been addressed.

Comment (AmrSaleh2234) [inline modules/ProcedureSetting/Requests/CreateProcedureSettingRequest.php:31]: The PR description doesn't mention this significant change to the request validation. Please update the description to clearly explain the addition of the required `parent_id` field and its implications.

Comment (AmrSaleh2234) [inline modules/ProcedureSetting/Requests/CreateProcedureSettingRequest.php:31]: There's no indication in the PR description about how this change was tested. Please add testing details including how parent/child relationships were validated and edge cases were handled.

Comment (AmrSaleh2234) [inline modules/ProcedureSetting/Requests/CreateProcedureSettingRequest.php:31]: The change introduces a new required field but doesn't show any corresponding migration or database changes. Please confirm the database schema has been updated to support this new relationship.

Comment (AmrSaleh2234) [inline modules/ProcedureSetting/Requests/CreateProcedureSettingRequest.php:31]: Consider the performance implications of adding a required parent relationship - will this cause N+1 query problems when fetching procedure settings? Please confirm this has been addressed with proper eager loading.

Comment (AmrSaleh2234) [inline modules/ProcedureSetting/Requests/CreateProcedureSettingRequest.php:31]: The validation doesn't specify what happens if someone tries to create a procedure setting with a parent that belongs to a different workflow. Consider adding validation to ensure parent and child belong to the same workflow if that's a business requirement.

Comment (AmrSaleh2234) [inline modules/ProcedureSetting/Requests/CreateProcedureSettingRequest.php:31]: There's no validation to prevent deep nesting of procedure settings which could lead to performance issues. Consider adding a maximum depth validation if applicable to your business rules.

Comment (AmrSaleh2234) [inline modules/ProcedureSetting/Requests/CreateProcedureSettingRequest.php:31]: The change doesn't show any updates to the ProcedureSetting model to define this new relationship. Please confirm the model has been updated with the proper `belongsTo` relationship.

File: modules/ProcedureSetting/Resources/routes/api.php [hunk 40]
```diff
@@ -17,6 +18,14 @@
     Route::put('/{id}', [ProcedureSettingController::class, 'update']);
     Route::delete('/{id}', [ProcedureSettingController::class, 'delete']);
 
+    // Internal Procedure Settings (child procedure settings with form key)
+    Route::get('/internal-procedures', [InternalProcedureSettingController::class, 'index']);
```
Comment (AmrSaleh2234) [inline modules/ProcedureSetting/Resources/routes/api.php:22]: The route `/internal-procedures` should be more specific since it's a child resource. Consider prefixing it with `/{id}/` to make the relationship clearer.

File: modules/ProcedureSetting/Resources/routes/api.php [hunk 41]
```diff
@@ -17,6 +18,14 @@
...
+    Route::get('/{id}/available-forms', [InternalProcedureSettingController::class, 'availableForms']);
+    Route::get('/{id}/internal-procedures/by-form/{formKey}', [InternalProcedureSettingController::class, 'showByForm']);
+    Route::post('/internal-procedures', [InternalProcedureSettingController::class, 'store']);
```
Comment (AmrSaleh2234) [inline modules/ProcedureSetting/Resources/routes/api.php:25]: The POST route `/internal-procedures` should also include the parent ID in the path to maintain consistency with other routes, e.g. `/{id}/internal-procedures`.

File: modules/ProcedureSetting/Resources/routes/api.php [hunk 42]
```diff
@@ -17,6 +18,14 @@
...
+    Route::put('/{id}/internal-procedures/{internalProcedureId}', [InternalProcedureSettingController::class, 'update']);
```
Comment (AmrSaleh2234) [inline modules/ProcedureSetting/Resources/routes/api.php:26]: The parameter naming is inconsistent - `internalProcedureId` should be consistent with other routes (either use `id` or `internalProcedureId` throughout).

File: modules/ProcedureSetting/Resources/routes/api.php [hunk 43]
```diff
@@ -17,6 +18,14 @@
...
+    Route::delete('/{id}/internal-procedures/{internalProcedureId}', [InternalProcedureSettingController::class, 'destroy']);
```
Comment (AmrSaleh2234) [inline modules/ProcedureSetting/Resources/routes/api.php:27]: Same as above - parameter naming inconsistency (`internalProcedureId` vs `id`). Also consider if `destroy` should be aligned with `delete` used in parent resource routes.

File: modules/Process/Models/Process.php [hunk 44]
```diff
@@ -44,7 +44,7 @@ protected static function boot(): void
 
         Relation::morphMap([
             'client_request'        => \Modules\ClientRequest\Models\ClientRequest::class,
-            'employee_task_request' => \Modules\EmployeeTask\Models\EmployeeTaskRequest::class,
+            'employee_task' => \Modules\EmployeeTask\Models\EmployeeTaskRequest::class,
```
Comment (AmrSaleh2234) [inline modules/Process/Models/Process.php:47]: The key `'employee_task'` doesn't match the pattern used for other morph map entries (like `'client_request'`). Consider using consistent naming convention, either:
1. Keep the `_request` suffix: `'employee_task_request'`
2. Or update all entries to remove `_request` suffix for consistency

File: modules/Shared/InternalProcessType/Enums/InternalProcessForm.php [hunk 45]
```diff
@@ -72,6 +72,39 @@ public function toDefinition(): array
...
 
+    /**
+     * Returns the procedure-setting category types this form is applicable to.
+     * Matched against ProcedureSetting::type (ProcedureSettingType values).
+     *
+     * @return list<string>
+     */
+    public function applicableTypes(): array
+    {
+        return match ($this) {
```
Comment (AmrSaleh2234) [inline modules/Shared/InternalProcessType/Enums/InternalProcessForm.php:83]: Consider using `ProcedureSettingType` constants instead of hardcoded strings for better maintainability and type safety.

File: modules/Shared/InternalProcessType/Enums/InternalProcessForm.php [hunk 46]
```diff
@@ -72,6 +72,39 @@ public function toDefinition(): array
...
+    /**
+     * @param string $procedureType  A ProcedureSettingType::value
+     * @return list<self>
+     */
+    public static function forType(string $procedureType): array
+    {
+        return array_values(
+            array_filter(
+                self::cases(),
+                static fn (self $form): bool => in_array($procedureType, $form->applicableTypes(), true),
```
Comment (AmrSaleh2234) [inline modules/Shared/InternalProcessType/Enums/InternalProcessForm.php:103]: The `array_filter` + `array_values` combination could be simplified to just `array_filter` since we're not reindexing the array numerically.
